third_party/shaka-player/shaka-player.compiled.js

```diff
@@ -0,0 +1,493 @@
+(function(){var innerGlobal=typeof window!="undefined"?window:global;var exportTo={};(function(window,global){var n,aa="function"==typeof Object.defineProperties?Object.defineProperty:function(b,c,d){b!=Array.prototype&&b!=Object.prototype&&(b[c]=d.value)},ba="undefined"!=typeof window&&window===this?this:"undefined"!=typeof global&&null!=global?global:this;function ca(){ca=function(){};ba.Symbol||(ba.Symbol=da)}var da=function(){var b=0;return function(c){return"jscomp_symbol_"+(c||"")+b++}}();
```

Comment:
Keeping compiled code in the repo is a sync nightmare.

This also doesn't allow us to tree-shake it properly.

Is there any way we can work with the Shaka team to refactor in order to tree-shake better? This is incredibly large, and I feel like for adaptive playback may be counterintuitive, since it needs to download a larger amount of JS than the first few, lossy seconds of playback.

Keeping it concatenated into the extension also forces cold-downloads, since the shaka binary is then always tied to the extension binary (which is always forcibly refreshed on every release, weekly.) So you'd end up in the worst loading case fairly frequently.

Reply:
We can definitely clone or use the shaka npm package and build in tree such that the unminimized version is available to your own internal packager. I just modeled this after other js packages I saw.

Unfortunately adaptive streaming is a complex beast and Shaka is still the smallest of all other players at the time I wrote this. Someone could probably write a micro library that covers most of the hls spec and only supports .mp4 files, but it'd still be ~100kb uncompressed I'd guess.